source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -380,6 +382,20 @@ int StreamHandleWrapper::luaHeaders(lua_State* state) {
   return 1;
 }
 
+void StreamHandleWrapper::onBeforeFinalizeUpstreamSpan(
+    Tracing::Span& span, const Http::ResponseHeaderMap* response_headers, bool success) {
+  if (success && response_headers != nullptr) {
+    uint64_t status_code{};
+    if (!absl::SimpleAtoi(response_headers->getStatusValue(), &status_code)) {
+      span.setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
+    }
+    span.setTag("lua_http_call_http_status",
+                Http::CodeUtility::toString(static_cast<Http::Code>(status_code)));
+  } else {
+    span.setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
+  }
+}
```

Comment:
As this is new tracing, can we just remove all of this and have this done by the base async client code? Per other comments it should set error for us, and I think there is a common http-status tag that should be set or we can set it if not?

Reply:
I have removed this. However, I think we can skip on adding `http.status` for now since it is probably redundant with `http.status_code`; until someone needs it.